### DISCUSSION TITLE
Atlantis: One Item Menu not being auto clicked.

### DISCUSSION THREAD

**Discussion by tony-in-nz** [primefaces/community#1322] [PrimeFaces Templates]
Hi Team,

>> mert.sincan

With reference to this I checked in 6.0.0 to see if it had been added and found it had not yet.

https://forum.primefaces.org/viewtopic.php?f=48&t=79329

I added the code manually; but it has no effect (previously I assumed it was my fault it was not working so waited until 6.0.0 came out).

**Comment by tony-in-nz** [comment]
```
		<myfaces.version>4.0.2</myfaces.version>
		<primefaces.version>12.0.7</primefaces.version>
		<fontawesome.version>6.2.0</fontawesome.version>
		<version.com.primefaces.atlantis>6.0.0</version.com.primefaces.atlantis>
```

**Reply by melloware** [reply]
@tonyherstell so in the latest Atlantis theme you are not seeing this code patched?

```js
_bindEvents: function() {
        var $this = this;

        this.tabMenu.find('.menu-button').off('click.menu').on('click.menu', function(e) {
            $this.wrapper.removeClass('layout-wrapper-menu-active');
            $this.tabMenuNavLinks.removeClass('active-item'); /* PLEASE ADD THIS LINE */
            if(!$this.isOverlayMenu()) {
                $(window).trigger('resize');
            }

            e.preventDefault();
        });
```
??

**Reply by melloware** [reply]
When I am in the Atlantis Demo and I click the Buy Now link I only have to click it once?  So I am not sure what you mean by click "twice"?

![image](https://github.com/primefaces/community/assets/4399574/6efc0bdb-5c50-4f64-93fe-546917d77547)

**Comment by tony-in-nz** [comment]
I have tried; but I must have messed it up as there was no effect.
I will try again tonight
Is the job market really bad where you are?

See thee menus:
https://grangeroadstables.nz/

Making a top level menu item capable of triggering an action/page throw may
be the best option; but may require a lot of changes to the menu sections.

**Comment by melloware** [comment]
Ahh I see what you mean let me take a look and see if I can send you a
patch.

On Mon, Apr 29, 2024 at 7:40 PM Tony ***@***.***> wrote:

> I have tried; but I must have messed it up as there was no effect.
> I will try again tonight
> Is the job market really bad where you are?
>
> See thee menus:
> https://grangeroadstables.nz/
>
> Making a top level menu item capable of triggering an action/page throw
> may
> be the best option; but may require a lot of changes to the menu sections.
>
>
>
> On Mon, 29 Apr 2024 at 03:24, Melloware ***@***.***> wrote:
>
> > Ahhh you are right I don't see it either in Atlantis 6.0 simply patch
> the
> > line yourself?
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/orgs/primefaces/discussions/1322#discussioncomment-9252904>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AB6LSIWBZYBSLWVOFYJMXW3Y7UIBZAVCNFSM6AAAAABD7JF64SVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TENJSHEYDI>
>
> > .
> > You are receiving this because you were mentioned.Message ID:
> > ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/primefaces/discussions/1322#discussioncomment-9268521>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBSDVRRH4IM4GEVR2SA4M3Y73K6VAVCNFSM6AAAAABD7JF64SVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TENRYGUZDC>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
==============================
Melloware
***@***.***
http://melloware.com
==============================

**Reply by melloware** [reply]
OK so this is a little tricker than just a simple fix. Basically you want this...

```xml
<pa:tab id="tab6" icon="pi pi-calendar" title="Open Calendar"></pa:tab>
```

Where you just have a tab and want it to navigate.

![image](https://github.com/primefaces/community/assets/4399574/50196fff-a339-401b-be0f-331e26025bdd)

The Tab is meant to open a menu item and not navigate.  I think you need to override the `openMenu` method instead of opening and closing the drawer to perform  your navigation.

```js
    openMenu: function(link, restore) {
        this.sidebar.css('overflow','hidden');
        var parent = link.parent();
        parent.addClass('active-item');

        this.wrapper.addClass('layout-wrapper-menu-active');
        if(restore) {
            this.wrapper.addClass('layout-wrapper-menu-active-restore');
        }

        this.tabMenu.find('.layout-tabmenu-content').removeClass('layout-tabmenu-content-active').
                eq(parent.index()).addClass('layout-tabmenu-content-active');

        if(this.cfg.stateful) {
            this.saveTabState(parent.attr('id'));
        }
    }
```

**Comment by tony-in-nz** [comment]
I am starting a new project and will try this out over the weekend.

**Comment by tandraschko** [comment]
already fixed in newest release
